test_regress/t/t_vpi_dump.cpp

```diff
@@ -0,0 +1,204 @@
+    std::vector<int32_t> instance_options = {
+        vpiNet,
+        vpiNetArray,
+        vpiReg,
+        vpiRegArray,
+    };
+
+    std::vector<int32_t> module_options = {
+        vpiModule,  // Aldec SEGV on mixed language
+        // vpiModuleArray,       // Aldec SEGV on mixed language
+        // vpiIODecl,            // Don't care about these
+        vpiMemory, vpiIntegerVar, vpiRealVar,
+        // vpiRealNet, Vpi extension
+        vpiStructVar, vpiStructNet, vpiNamedEvent, vpiNamedEventArray, vpiParameter,
+        // vpiVariables, // parent of vpiReg, vpiRegArray, vpiIntegerVar, etc vars
+        // vpiSpecParam,         // Don't care
+        // vpiParamAssign,       // Aldec SEGV on mixed language
+        // vpiDefParam,          // Don't care
+        vpiPrimitive, vpiPrimitiveArray,
+        // vpiContAssign,        // Don't care
+        vpiProcess,  // Don't care
+        vpiModPath, vpiTchk, vpiAttribute, vpiPort, vpiInternalScope,
+        // vpiInterface,         // Aldec SEGV on mixed language
+        // vpiInterfaceArray,    // Aldec SEGV on mixed language
+    };
+
+    // append base class vpiInstance members
+    module_options.insert(module_options.begin(), instance_options.begin(),
+                          instance_options.end());
+
+    std::vector<int32_t> struct_options = {
+        vpiNet,       vpiReg,       vpiRegArray,       vpiMemory,
+        vpiParameter, vpiPrimitive, vpiPrimitiveArray, vpiAttribute,
+        vpiMember,
+    };
+
+    return decltype(iterate_over){
+        {vpiModule, module_options},
+        {vpiInterface, instance_options},
+        {vpiGenScope, module_options},
+
+        {vpiStructVar, struct_options},
+        {vpiStructNet, struct_options},
+
+        {vpiNet,
+         {
+             // vpiContAssign,        // Driver and load handled separately
+             // vpiPrimTerm,
+             // vpiPathTerm,
+             // vpiTchkTerm,
+             // vpiDriver,
+             // vpiLocalDriver,
+             // vpiLoad,
+             // vpiLocalLoad,
+             vpiNetBit,
+         }},
+        {vpiNetArray,
+         {
+             vpiNet,
+         }},
+        {vpiRegArray,
+         {
+             vpiReg,
+         }},
+        {vpiMemory,
+         {
+             vpiMemoryWord,
+         }},
+        {vpiPort,
+         {
+             vpiPortBit,
+         }},
+        {vpiGate,
+         {
+             vpiPrimTerm,
+             vpiTableEntry,
+             vpiUdpDefn,
+         }},
+        {vpiPackage,
+         {
+             vpiParameter,
+         }},
+    };
+}();
+
+void modDump(vpiHandle it, int n) {
+
+    if (n > 8) {
+        printf("going too deep\n");
+        return;
+    }
+
+    while (TestVpiHandle hndl = vpi_scan(it)) {
+        for (int i = 0; i < n; i++) printf("    ");
+        auto type = vpi_get(vpiType, hndl);
+        auto name = vpi_get_str(vpiName, hndl);
+        auto fullname = vpi_get_str(vpiFullName, hndl);
```

Comment:
Please avoid using auto here as easier to read when see the return types.  (Typically we use auto only for iterators where it doesn't really matter the type involved)